src/OpenTK.Windowing.GraphicsLibraryFramework/Input/KeyboardState.cs

```diff
@@ -233,30 +169,25 @@ public override string ToString()
             return builder.ToString();
         }
 
-        // This shouldn't be necessary but I'll keep it in just in case.
-        [Conditional("DEBUG")]
-        private static void ValidateOffset(int offset)
+        internal void Update()
         {
-            if (offset < 0 || offset >= NumInts * IntSize)
-            {
-                throw new ArgumentOutOfRangeException();
-            }
+	        Utils.Swap(ref _keys, ref _keysPrevious);
```

Comment:
I would just write this as a simple `temp` variable swap. It's way more obvious to the average programmer what is going on.  This really isn't a big deal either way. If you like it, don't change it.

Reply:
I'd already put a Swap method in JoystickInput, this just makes it generally available. If someone wants to see what it's doing, they can Ctrl+Click in any IDE worth using.